testdata/issues/issue165.wit

```diff
@@ -0,0 +1,18 @@
+// https://github.com/bytecodealliance/wasm-tools-go/issues/165
+package issues:issue165;
+
+interface types {
+	resource z {
+		constructor(a: f64);
+	}
+}
+
+world resources {
+	use types.{z};
+
+	record big {
+		x1: borrow<z>,
+	}
+
+	export big-record: func(r: big);
```

Comment:
Found another casting issue:

If you add `export maybe-with-z: func(a: option<borrow<z>>);` as another export, the generated Go code would error out, saying
```
cannot convert cm.Some[cm.Rep](cm.Reinterpret[cm.Rep]((uint32)(f1))) (value of type cm.Option[cm.Rep]) to type cm.Option[types.Z]
```